src/functions/commerce/provider.php

```diff
@@ -93,12 +87,67 @@ function tec_tribe_commerce_is_available() {
 		);
 	}
 
+	$should_be_active    = tec_tribe_commerce_should_be_active();
+	$should_be_available = $available && $should_be_active;
+
 	/**
 	 * Filter whether we should disable TribeCommerce PayPal or not.
 	 *
 	 * @since 5.1.10
 	 *
 	 * @param boolean $available should be available or not.
 	 */
-	return apply_filters( 'tec_tribe_commerce_is_available', $available );
+	return apply_filters( 'tec_tribe_commerce_is_available', $should_be_available );
+}
+
+/**
+ * Check if TribeCommerce should be active or not.
+ *
+ * @since TBD
+ *
+ * @return bool
+ */
+function tec_tribe_commerce_should_be_active() {
+
+	// If new install then just return false.
+	if ( tribe_installed_after( 'Tribe__Tickets__Main', '5.1.10' ) ) {
+		return false;
+	}
+
+	$should_be_active = tec_tribe_commerce_has_active_tickets();
+
+	/**
+	 * Filter whether TribeCommerce should be active or not.
+	 *
+	 * @since TBD
+	 *
+	 * @param boolean $should_be_active Should TribeCommerce be kept activated or not.
+	 */
+	return apply_filters( 'tec_tribe_commerce_should_be_active', $should_be_active );
+}
+
+/**
+ * Check if the site has created tickets using TribeCommerce.
+ *
+ * @since TBD
+ *
+ * @param bool $force_check Setting this as true will bypass the cached/default value.
+ *
+ * @return bool
+ */
+function tec_tribe_commerce_has_active_tickets( $force_check = false ) {
+
+	$cache_key = 'tec_tribe_commerce_has_active_tickets';
+
+	if ( ! $force_check ) {
+		// If we don't find any cached value, we keep things activated.
+		return (bool) get_option( $cache_key, true );
+	}
+
+	$has_active_tickets = (bool) tribe_tickets()->by( 'post_type', 'tribe_tpp_tickets' )->where( 'is_active' )->count();
+
+	// Cache the data.
+	update_option( $cache_key, $has_active_tickets );
```

Comment:
use a transient with a 24h expiration here.

Reply:
I have avoided using transient intentionally.

We are setting this value only when the customer is visiting the `Tickets` settings page. So, this is getting updated whenever the admin is actively using the website or trying to change the settings, which makes it realtime.

Using transient will not make it realtime and we might end up not showing the notices even if there are no active tickets.

Reply:
Also, `get_transient` doesn't have any default value option like `get_option`. So, changing to that will require some logic updates around `tec_tribe_commerce_should_be_active`

Reply:
This is a transient piece of data, meaning it changes once certain piece of criteria is met. The same things you do with `update_option` and `get_option` can be used with the transient, the diff is lets say our QA needs to reset this check for some reason, they can just use any simple transient manager Like they are already used to.